Lesson17/Main.java

```diff
@@ -0,0 +1,107 @@
+package Lesson17;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class Main {
+    public static void main(String[] args) {
+
+        /* Tasks 1: Створити колекцію записати туда 100 рандомних значень.
+               - Знайти суму непарних чисел у колекції.
+               - Порахувати кількість парниих чисел
+               - Тільки позитивні числа цієї колекції записати в нову колекцію
+         */
+        
+        List<Integer> list = new ArrayList<>();
+        for (int i = 0; i <= 100; i++) {
+            list.add(new Random().nextInt(100));
+        }
+        Iterator<Integer> iter = list.iterator(); while(iter.hasNext()) {
+            Integer number = iter.next();
+            System.out.println(number);        // Виведення первинного вигляду Колекції рандомних чисел
+        }
+
+        int oddNumbersSum = list.stream()
+                .filter(number -> (number % 2) != 0)
+                .mapToInt(number -> number)
+                .sum();
+        System.out.println("Сума непарних чисел: " + oddNumbersSum);
+
+        long evenNumbers = list.stream()
+                .filter(number -> (number % 2) == 0)
+                .count();
+        System.out.println("Кількість парних чисел: " + evenNumbers);
+
+        List<Integer> listOfPositiveNumbers = list.stream()
+                .filter(number -> (number > 0)).toList();
+
+        System.out.println("Колекція позитивних чисел: " + listOfPositiveNumbers);
+
+
+        // Task 2: Оголосити довільний масив рядків. Відсортувати елементи потоку за довжиною слова. Зберегти у колекцію.
+        String str = "The forEach метод запускає внутрішню ітерацію елементів у потоці та підраховує елементи, які задовольняють задану умову";
+        str = str.replaceAll("[^a-zA-Zа-яА-ЯіІїЇєЄҐґ ]", "");
+        String[] list2 = str.split(" ");
+
+        Comparator<String> comparator = (i1, i2) -> (i1.length() - i2.length());
+
+        List<String> sortedStringArray = Arrays.stream(list2)
+                .sorted(comparator)
+                .toList();
+        System.out.println("Відсортовані за довжиною слова: " + sortedStringArray);
```

Comment:
Коли компаратор використовуэться в одному місці то його краще зразу там і писати, ще можна використи метод референс і трішки його спростити
```suggestion
        List<String> sortedStringArray = Arrays.stream(list2)
                .sorted(Comparator.comparingInt(String::length))
                .toList();
        System.out.println("Відсортовані за довжиною слова: " + sortedStringArray);
```

Reply:
Виконано, дякую за підказку :)